### ISSUE TITLE
Support acknowledgement responses to user interactions

### ISSUE THREAD

**Issue by AndrewEckart** [nicklambourne/slackblocks#3]
When [responding to interactions](https://api.slack.com/messaging/interactivity#responding_to_interactions), it's necessary for the API server which receives the interaction to construct an [acknowledgement response](https://api.slack.com/interactivity/handling#acknowledgment_response). The response payload can be a message, formatted in the usual way (i.e. `{"blocks": [..]}`, but this message doesn't require a `channel` property and can also include a `replace_original: true` attribute to replace the message which was interacted with.

Proposal is to make `channel` optional, or create a base class `BaseMessage` which is subclassed for regular messages and acknowledgement responses.

**Comment by nicklambourne** [comment]
Hey Andrew, apologies for the delayed response to this, I've been a bit busy with research and thesis work. I like the proposal and will aim to have it implemented in the next day or two.

**Comment by nicklambourne** [comment]
Working on this now. I forgot to mention that as a stopgap I'd made the channel field optional on Message for the time being.

**Comment by nicklambourne** [comment]
I think I've implemented what is was you're after. It's difficult to run an integration test for this kind of thing without setting up a sever. Let me know if I've overlooked anything.

**Comment by AndrewEckart** [comment]
Looks good! I'll test it out when I get a chance.

It looks like I slightly misread the [different ways](https://api.slack.com/interactivity/handling#responses) Slack allows an API server to respond to an interaction. Apparently an "Acknowledgement response" is just a plain HTTP request with a 200 status, which must be sent within 3 seconds (this just prevents a little error icon from appearing next to the button the user clicked in Slack).

What you implemented, and what I originally thought of as an acknowledgement response, actually has the very generic name of "Message response". Besides not needing a channel (Slack can figure that out from the response URL it gives you), it's basically a typical message with two additional attributes: the boolean `replace__original`, which you already implemented, and `response_type`, which can be set to `"ephemeral"` to send a message only visible to the interacting user.

TL:DR; You can add `response_type` to the `AcknowledgementResponse` constructor (or just add an `ephemeral` boolean argument since that's the only value it can take anyway). You could also rename `AcknowledgementResponse` to `MessageResponse`, although imo Slack's terminology is a little confusing. Lastly, the default value for `replace_original` should probably be False instead of True, but it's not a big deal. Thanks for taking the time to implement it!

**Comment by nicklambourne** [comment]
That makes a lot of sense wrt the availability of documentation for AcknowledgementResponses. I've made the suggested changes in the latest version.
